engine/classes/Elgg/Filesystem/AdapterService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+namespace Elgg\Filesystem;
+use Elgg\Filesystem\Adapter\Adapter;
+
+
+class AdapterService {
+	private $logger;
+	
+	/**
+	 *
+	 * @var string
+	 */
+	private $default;
+	
+	/**
+	 *
+	 * @var Adapter[]
+	 */
+	private $adapters;
+	
+	public function __construct(\Elgg\Logger $logger) {
+		$this->logger = $logger;
+	}
+	
+	public function set($name, Adapter $adapter, $default = false) {
+		if ($this->has($name)) {
+			throw new \UnexpectedValueException("Filesystem adapter \"$name\" already exists.");
+		}
+		$this->adapters[$name] = $adapter;
+		
+		if ($default) {
+			$this->setDefault($name);
+		}
+		return true;
+	}
+	
+	/**
+	 * 
+	 * @param type $name
+	 * @return Adapter
+	 * @throws \InvalidArgumentException
+	 */
+	public function get($name) {
+		if (!$this->has($name)) {
+			throw new \InvalidArgumentException("Filesystem Adapter \"$name\" does not exist.");
+		}
+		
+		return $this->adapters[$name];
+	}
+	
+	public function has($name) {
+		return isset($this->adapters[$name]);
+	}
+	
+	public function remove($name) {
```

Comment:
Need to actually merge them in first :). I will make an extra effort to get
them in this week.

On Mon, Apr 6, 2015 at 12:15 PM Brett Profitt <notifications@github.com>
wrote:

> In engine/classes/Elgg/Filesystem/AdapterService.php
> <https://github.com/Elgg/Elgg/pull/8092#discussion_r27821409>:
>
> > +	 * @return Adapter
> > +	 * @throws \InvalidArgumentException
> > +	 */
> > +	public function get($name) {
> > +		if (!$this->has($name)) {
> > +			throw new \InvalidArgumentException("Filesystem Adapter \"$name\" does not exist.");
> > +		}
> > +		
> > +		return $this->adapters[$name];
> > +	}
> > +	
> > +	public function has($name) {
> > +		return isset($this->adapters[$name]);
> > +	}
> > +	
> > +	public function remove($name) {
>
> I just saw those classes this morning. It's a shame they're not used in
> core.
>
> I'll be able to switch over to ArrayCollection assuming plugins are fully
> init'd by the time the service is called.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/8092/files#r27821409>.
>